handle double backslash at end of quoted string

cshorler opened the pull request:
fixes issues like these
```
puts "hello world\\"
puts "\\"
```

behaviour for escaped quote should be preserved as-is.

tested interactively.

bitwisecook commented:
@cshorler LGTM, merged, should i do a release or should i wait a little?

cshorler commented:
I noticed some other things as well:

1. When you increased the vscode version dependency and the typescript
version, it invalidated my build tree.  Npm install wasn't enough.

a.  Makefile target dist-clean is conventionally named "distclean" (one
word).  Up to this point I was manually cleaning, thinking the target
didn't exist... Requiring closer examination of the Makefile.

b.  Updating the typescript version seemed to be the (spurious) cause of
missing file errors and inferred tsconfig errors.  This was time wasted.

c.  package.json has a deprecated name for the vsce package.

d.  Launching the extension (debug) silently fails with the wrong vscode
version. (another reason not to insist on the latest).  I isolated
environments upgrading the core editor version may not be as trivial as it
is with Internet access.

Some of these points could be addressed before release. I'm not seeing any
other formatting issues in my code base yet.

Thanks for acting quickly last time - really Agile!

On Mon, 20 Feb 2023, 23:33 James Deucker, ***@***.***> wrote:

> @cshorler <https://github.com/cshorler> LGTM, merged, should i do a
> release or should i wait a little?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitwisecook/vscode-tcl/pull/21#issuecomment-1437668767>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHKKMFCXVRJZL3QSYLAK7DWYP5LJANCNFSM6AAAAAAVCMBO4U>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

cshorler commented:
@bitwisecook

bitwisecook commented:
Cool I'll try to get this merged up ASAP and I'll take a look at the above

bitwisecook commented:
0.4.3 is out

i backed the engine down to 1.74.0, I don't like having to too far behind because eventually i have to bots complaining about vulns, so i just tend to update everything each release.

i renamed dist-clean -> distclean

i updated to the current pacakge name for vscode

cshorler commented:
Awesome!

On Wed, 22 Feb 2023, 19:25 James Deucker, ***@***.***> wrote:

> 0.4.3 is out
>
> i backed the engine down to 1.74.0, I don't like having to too far behind
> because eventually i have to bots complaining about vulns, so i just tend
> to update everything each release.
>
> i renamed dist-clean -> distclean
>
> i updated to the current pacakge name for vscode
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitwisecook/vscode-tcl/pull/21#issuecomment-1440670642>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHKKMAZFG2XGLT5CASYSDDWYZRZRANCNFSM6AAAAAAVCMBO4U>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
